### PR TITLE
Incorporates limiter into the Implicit Stribeck solver

### PR DESCRIPTION
**Pull request by amcastro-tri**
In this PR `ImplicitStribeckSolver` starts using `DirectionChangeLimiter`. The impact problem unit tests along with this PR would not succeed without this change (the NR solver does need the limiter in this case).

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)

### CODE REVIEW

**Comment by amcastro-tri** [comment]
+@edrumwri for feature review please.

---

Review status: all discussions resolved, platform LGTM missing, commit history needs curation

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995#-:-LEypCCR-oIqDfREqej9:b-a98c4g)*

**Comment by edrumwri** [comment]
Checkpoint.

---

Reviewed 2 of 3 files at r1.
Review status: 9 unresolved discussions, LGTM missing from assignee edrumwri, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h, line 696 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz7NwIDCDqij-NmFjI:-LEz7NwIDCDqij-NmFjJ:b-qsgbnm) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h#L696)):*
> ```objc
>   // all contact points. The angle change θ is defined by the dot product
>   // between vₜᵏ⁺¹ and vₜᵏ as: cos(θ) = vₜᵏ⁺¹⋅vₜᵏ/(‖vₜᵏ⁺¹‖‖vₜᵏ‖).
>   // We'll do so by computing a coefficient 0 < α < 1 so that if the
> ```

nit <= 1

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h, line 733 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz7r4K5ghYn_XkrWHY:-LEz7r4K5ghYn_XkrWHZ:b-ygtm91) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h#L733)):*
> ```objc
> 
>   // Precomputed value of cos(theta_max), used by DirectionChangeLimiter.
>   double cos_theta_max_{std::cos(parameters_.theta_max)};
> ```

Should this be `const double`or do you anticipate that it might be changed?

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc, line 476 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz8eMj-SUpYZsRIuaO:-LEz8eMj-SUpYZsRIuaP:b-vutq5h) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc#L476)):*
> ```cpp
>   T alpha = 1.0;
>   double v_stribeck = parameters_.stiction_tolerance;
>   for (int ic = 0; ic < nc_; ++ic) {
> ```

Nit: please explain `ic` and `ik` like you did elsewhere since these variable names are not descriptive.

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc, line 486 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz8TcjBSoGL5-3C5Pj:-LEz8TcjBSoGL5-3C5Pk:bfy49b0) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc#L486)):*
> ```cpp
>             cos_theta_max_, v_stribeck, parameters_.relative_tolerance));
>   }
>   return alpha;
> ```

This function has a post-condition that alpha > 0. Please assert that.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 643 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzPbl_BgfiUHvmmVEb:-LEzPbl_BgfiUHvmmVEc:b-ar4pom) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L643)):*
> ```cpp
> // This test verifies that the implicit stribeck solver can correctly predict
> // transitions in a problem with impact. This is a 2D problem consisting of a
> // cylinder of radius R, mass m, and rotational inertia I, initially at height
> ```

Talking about a cylinder in 2D doesn't make sense (I'm guessing you're thinking of a 2D projection of a 3D problem). What is the appropriate shape for 2D?

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 647 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz93HG8XaQx9uPoBYy:-LEz93HG8XaQx9uPoBYz:b-jasdi9) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L647)):*
> ```cpp
> // At t = 0, the cylinder is given an initial horizontal velocity vx0 and zero
> // vertical velocity vy0 = 0. The cylinder will undergo a parabolic free flight
> // towards the ground until the moment of impact at which the vertical velocity
> ```

Without reading further,  this seems like you're setting up an algebraic impact problem- rephrasing slightly, you say that the vertical velocity goes to zero at the moment of impact- while the compliant model that I'm assuming you're using does not actually predict a "moment" of impact as much as an interval of impact.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 650 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzNOBr0tswCjQJOfOW:-LEzNOBt7jFIXWV4slgW:bucwins) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L650)):*
> ```cpp
> // goes to zero (a purely inelastic collision). Since we know the initial
> // height, we therefore know that the vertical velocity at the time of impact
> // will be vy = -sqrt(m⋅g⋅h₀), with g the acceleration of gravity. Therefore the
> ```

The vertical velocity at the time of impact is vy = -g*sqrt(2*h/g)

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 655 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzOsiOBdf_Ooc9nNKW:-LEzOsiOBdf_Ooc9nNKX:b-26x0wz) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L655)):*
> ```cpp
> // The implicit Stribeck solver needs to know the normal forces in advance. We
> // compute the normal force in order to exactly bring the cylinder to a stop in
> // the vertical direction within a time interval dt. That is, we set the normal
> ```

"within a time interval dt" from the time that the bodies first make contact, yes?

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 656 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzP18G7yrcMtK7EtAD:-LEzP18G7yrcMtK7EtAE:bsva4wc) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L656)):*
> ```cpp
> // compute the normal force in order to exactly bring the cylinder to a stop in
> // the vertical direction within a time interval dt. That is, we set the normal
> // force to fn = -m * vy / dt + m * g, where the small contribution due to
> ```

Wouldn't this need to have m * g / dt?

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995#-:-LEzQJRpDkStPaJwh3WZ:by96gdd)*

**Comment by edrumwri** [comment]
Review status: 9 unresolved discussions, LGTM missing from assignee edrumwri, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 650 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzNOBr0tswCjQJOfOW:-LEzssBUA_3QANFIpdFq:bgvw8ic) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L650)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

The vertical velocity at the time of impact is vy = -g*sqrt(2*h/g)
</blockquote></details>

The asterisk formatted this wrong: it should be vy = -g⋅sqrt(2h/g)

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)*

**Comment by amcastro-tri** [comment]
Review status: 7 unresolved discussions, LGTM missing from assignee edrumwri, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h, line 733 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz7r4K5ghYn_XkrWHY:-LF-96w_9Ph49X3k4vwr:b-96taur) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.h#L733)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Should this be `const double`or do you anticipate that it might be changed?
</blockquote></details>

Yes. The only other place where a user could change this is in `ImplicitStribeckSolver::set_solver_parameters()`, where its value is updated.

---

*[multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc, line 486 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz8TcjBSoGL5-3C5Pj:-LF-9owBC5OdCq56VrRc:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/implicit_stribeck_solver.cc#L486)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

This function has a post-condition that alpha > 0. Please assert that.
</blockquote></details>

Done.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 643 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzPbl_BgfiUHvmmVEb:-LF-BpbH1tz6RswJAiKw:bb71pro) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L643)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Talking about a cylinder in 2D doesn't make sense (I'm guessing you're thinking of a 2D projection of a 3D problem). What is the appropriate shape for 2D?
</blockquote></details>

PTAL

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 647 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz93HG8XaQx9uPoBYy:-LF-A9fd6cOawjfBP00J:by1hk7n) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L647)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Without reading further,  this seems like you're setting up an algebraic impact problem- rephrasing slightly, you say that the vertical velocity goes to zero at the moment of impact- while the compliant model that I'm assuming you're using does not actually predict a "moment" of impact as much as an interval of impact.
</blockquote></details>

There is no compliance in the normal direction (not yet). The solver takes a precomputed value of the normal force (see `ImplicitStribeckSolver::SetProblemData()`) that could yes, come from a compliant model (as it'll happen in MBP) or, as in this case, we provided an analytical value that exactly drives the vertical velocity to zero. The solver does not know where the normal force comes from, it just uses it.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 650 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzNOBr0tswCjQJOfOW:-LF-CBDTCQ4ScHIYFDMg:bl9r7fi) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L650)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

The asterisk formatted this wrong: it should be vy = -g⋅sqrt(2h/g)
</blockquote></details>

Clearly neither of us passed our physics 101 class. It should be `vy = -sqrt(2⋅g⋅h0)` (as in the code). Conservation of energy (assuming no friction with the environment, of course, mass plays no role here). It is always a good thing to check for units.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 655 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzOsiOBdf_Ooc9nNKW:-LF-CorD9rWOOvdLg8Vk:b-o8xmcu) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L655)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

"within a time interval dt" from the time that the bodies first make contact, yes?
</blockquote></details>

Yes. As you very well know, this is an approximation to the impact problem (finite impulse in zero time) using a finite force over a very small period of time dt. Conservation of momentum still is true. Added your observation though.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 656 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzP18G7yrcMtK7EtAD:-LF-D7_qErAPHWg06_wT:bviwioa) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L656)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Wouldn't this need to have m * g / dt?
</blockquote></details>

nope. To see why (of course you can do the full math) in a blink of an eye, just take the limit to dt --> 0, why should g matter? it doesn't, it's an impact.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)*

**Comment by edrumwri** [comment]
:lgtm: - nice work dude!

---

Reviewed 3 of 3 files at r2.
Review status: 6 unresolved discussions, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 647 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEz93HG8XaQx9uPoBYy:-LF-GIXCCPDKnZPqdaUn:btdsfzr) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L647)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

There is no compliance in the normal direction (not yet). The solver takes a precomputed value of the normal force (see `ImplicitStribeckSolver::SetProblemData()`) that could yes, come from a compliant model (as it'll happen in MBP) or, as in this case, we provided an analytical value that exactly drives the vertical velocity to zero. The solver does not know where the normal force comes from, it just uses it.
</blockquote></details>

I see.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 650 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzNOBr0tswCjQJOfOW:-LF-Faib4k5LQUddRaYe:b2gm7kl) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L650)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

Clearly neither of us passed our physics 101 class. It should be `vy = -sqrt(2⋅g⋅h0)` (as in the code). Conservation of energy (assuming no friction with the environment, of course, mass plays no role here). It is always a good thing to check for units.
</blockquote></details>

Sorry dude, I passed:  g⋅sqrt(2h/g) = sqrt(2⋅g⋅h0)

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 656 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzP18G7yrcMtK7EtAD:-LF-m0Jt6BIjPJdYdW32:bgv4yat) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L656)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

nope. To see why (of course you can do the full math) in a blink of an eye, just take the limit to dt --> 0, why should g matter? it doesn't, it's an impact.
</blockquote></details>

Ah, silly me. Right, this is weird though because you're mixing impulsive and non-impulsive forces. It gives additional variables and logic to keep track off. 

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 658 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF--Biq9tdviMdWbW52:-LF--Biq9tdviMdWbW53:b-fvt52a) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L658)):*
> ```cpp
> // force to fn = -m * vy / dt + m * g, where the small contribution due to
> // gravity is needed to exactly bring the cylinder's vertical velocity to zero.
> // The equations governing the motion for the cylinder during impact are:
> ```

I'd point out again that pn is constant.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 660 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-0Hbq7jIHZi4XAkE7:-LF-0Hbq7jIHZi4XAkE8:b8zc4pq) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L660)):*
> ```cpp
> // The equations governing the motion for the cylinder during impact are:
> //   (1)  I⋅Δω = pt⋅R,  Δω  = ω, since ω0 = 0.
> //   (2)  m⋅Δvx = pt⋅R, Δvx = vx - vx0
> ```

Shouldn't this be m⋅Δvx = pt?

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 680 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-pgl48JiCzrGTH7wJ:-LF-pgl48JiCzrGTH7wK:b-3qa8c0) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L680)):*
> ```cpp
> //   pt = -m⋅vx0 / (1 + m⋅R²/I)
> // From Eq. (4), stiction occurs if vx0 is smaller than:
> //   vx0 ≤ vx_transition =  μ⋅(1 + m⋅R²/I)⋅pn/m
> ```

Please document vx_transition.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 693 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-pqjSCE-Ze2xnMvdl:-LF-pqjSCE-Ze2xnMvdm:b-kgqfhw) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L693)):*
> ```cpp
> 
>   // Computes tangential velocity Jacobian s.t. vt = Jt * v.
>   // Where vt is a vector in ℝ². Its firs component corresponds to the
> ```

sp: firs

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 746 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF33-bo2fySoETUW3R3:-LF33-bp-eTuGsiZkT9s:bbd9ebc) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L746)):*
> ```cpp
>   //   vy = -3.0 m/s (velocity at impact).
>   //   pn = 3.0 Ns
>   //   vx_transition = 0.6 m/s
> ```

How was this determined?

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995#-:-LF33dT-1P8deaP14Vdh:b-xxf8db)*

**Comment by amcastro-tri** [comment]
+@sherm1 for platform review please.

---

Review status: 6 unresolved discussions, platform LGTM missing, commit history needs curation

---



*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995#-:-LF35w7o6hej3xkCbYC2:b6vi582)*

**Comment by amcastro-tri** [comment]
Review status: 6 unresolved discussions, LGTM missing from assignee sherm1, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 650 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LEzNOBr0tswCjQJOfOW:-LF3DBAPCHUlBeB4VbsS:bno0fl4) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L650)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Sorry dude, I passed:  g⋅sqrt(2h/g) = sqrt(2⋅g⋅h0)
</blockquote></details>

you did!

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 658 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF--Biq9tdviMdWbW52:-LF3D_Fx56hKotjKbjB2:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L658)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

I'd point out again that pn is constant.
</blockquote></details>

Done.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 660 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-0Hbq7jIHZi4XAkE7:-LF3DdtC7z7ZRwhModll:buz1f0a) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L660)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Shouldn't this be m⋅Δvx = pt?
</blockquote></details>

True that.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 680 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-pgl48JiCzrGTH7wJ:-LF3E-kg5PUVmFXqYDLS:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L680)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

Please document vx_transition.
</blockquote></details>

Done.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 693 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-pqjSCE-Ze2xnMvdl:-LF3E79H8bgGsRSVBTUo:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L693)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

sp: firs
</blockquote></details>

Done.

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 746 at r2](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF33-bo2fySoETUW3R3:-LF3EmbC4riW-iBjnRia:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/951473120ecff03d1e82d816da4899b338fbd591/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L746)):*
<details><summary><i>Previously, edrumwri (Evan Drumwright) wrote…</i></summary><blockquote>

How was this determined?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)*

**Comment by edrumwri** [comment]
Review status: all discussions resolved, LGTM missing from assignee sherm1, platform LGTM missing, commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 660 at r1](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF-0Hbq7jIHZi4XAkE7:-LF3Lf_GCKGJ-KIXVtRT:bodkb50) ([raw file](https://github.com/robotlocomotion/drake/blob/be1430523e71f06682255858f4c00886b5559245/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L660)):*
<details><summary><i>Previously, amcastro-tri (Alejandro Castro) wrote…</i></summary><blockquote>

True that.
</blockquote></details>

Nit Ack! Now you've introduced a spurious comma.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)*

**Comment by sherm1** [comment]
Beautiful -- awesome unit test! Platform :lgtm: with one minor comment request.

---

Reviewed 2 of 3 files at r2, 1 of 1 files at r3.
Review status: 1 unresolved discussion, platform LGTM from [sherm1], commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 643 at r3](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF3H6jGDBj8BDt72Saz:-LF3H6jGDBj8BDt72Sb-:buziu5x) ([raw file](https://github.com/robotlocomotion/drake/blob/ecffaad3558a98005fd1da9fa67b283f8c60721f/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L643)):*
> ```cpp
> // This test verifies that the implicit stribeck solver can correctly predict
> // transitions in a problem with impact. A cylinder's COM is constrained to move
> // in the x-y plane with its axis aligned with the z-axis. Therefore, the
> ```

Minor: it took me several reads to figure out that _y_ is the vertical axis here. Most commonly the "x-y plane" would be the ground, with z up. (My mental picture at this point was that this was a cylinder resting on one of its circular end caps, which made the rest of the discussion hard to follow.) Even just a horizontal orientation for the cylinder is ambiguous -- I guess you mean to have the cylinder rolling, rather than sliding in the long direction. Please clarify the frame and cylinder orientation right away so that the rest makes sense on first read. Maybe call this a "rolling cylinder" if that's the right interpretation.

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995#-:-LF3MMUmB5rIWQmzyUo-:biy3mwk)*

**Comment by edrumwri** [comment]
Review status: 1 unresolved discussion, platform LGTM from [sherm1], commit history needs curation

---

*[multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc, line 643 at r3](https://reviewable.io/reviews/robotlocomotion/drake/8995#-LF3H6jGDBj8BDt72Saz:-LF3NF7_4OpwoX3LlpOq:bk9iy6) ([raw file](https://github.com/robotlocomotion/drake/blob/ecffaad3558a98005fd1da9fa67b283f8c60721f/multibody/multibody_tree/implicit_stribeck/test/implicit_stribeck_solver_test.cc#L643)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

Minor: it took me several reads to figure out that _y_ is the vertical axis here. Most commonly the "x-y plane" would be the ground, with z up. (My mental picture at this point was that this was a cylinder resting on one of its circular end caps, which made the rest of the discussion hard to follow.) Even just a horizontal orientation for the cylinder is ambiguous -- I guess you mean to have the cylinder rolling, rather than sliding in the long direction. Please clarify the frame and cylinder orientation right away so that the rest makes sense on first read. Maybe call this a "rolling cylinder" if that's the right interpretation.
</blockquote></details>

BTW I thought of suggested "rolling cylinder" too but rejected it because some people's definition of rolling (like Andy's, per personal communication) would include "rolling" (transitioning) from the circular contact manifold to the line contact manifold. But I have no strong opinion- I did also expect z to be "up" until our phone conversation clarified.  

---


*Comments from [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/8995)*
